src/createServer.js

```diff
@@ -1,3 +1,59 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}/`);
+    const originalText = normalizedUrl.pathname.slice(1);
+    const targetCase = normalizedUrl.searchParams.get('toCase');
+    const validCaseTypes = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+    res.setHeader('Content-Type', 'application/json');
+
+    const errors = [];
+
+    if (!originalText) {
+      errors.push({
+        message: 'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message: '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+      });
+    }
+
+    if (targetCase && !validCaseTypes.includes(targetCase)) {
+      errors.push({
+        message: 'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+      });
+    }
```

Comment:
Move it to the separate function
```suggestion
    const errors = ...();
```